dcos/mesos.py

```diff
@@ -198,28 +234,13 @@ def tasks(self, fltr="", completed=False):
 
         tasks = []
         for framework in self._framework_dicts(completed, completed):
-            tasks += \
-                [Task(task, self)
-                 for task in _merge(framework, *keys)
-                 if fltr in task['id'] or
-                 fnmatch.fnmatchcase(task['id'], fltr)]
+            for task in _merge(framework, keys):
```

Comment:
for speed?  a few cycles aren't going to cost us, and I'd rather not add more indirection